sqlglot/parser.py

```diff
@@ -1835,7 +1840,7 @@ def _parse_update(self) -> exp.Update:
             **{  # type: ignore
                 "this": self._parse_table(alias_tokens=self.UPDATE_ALIAS_TOKENS),
                 "expressions": self._match(TokenType.SET) and self._parse_csv(self._parse_equality),
-                "from": self._parse_from(modifiers=True),
+                "from": self._parse_from(joins=True),
```

Comment:
This is the only method where the `modifiers` argument is set to `True` in `_parse_from`. I don't think we need to actually parse _all_ modifiers here, but just the JOINs, hence the change.

FYI the only other "modifiers" in `Table` (which is parsed inside `_parse_from`) are `pivots` and `laterals`, but like I said, I doubt these would be useful in either `UPDATE` or `DELETE`. Can fix this in the future easily if it's a problem.